.travis.yml

```diff
@@ -1,14 +1,21 @@
 os:
-  - linux
-
+- linux
 language: go
-
 go:
-  - 1.9
-
+- 1.9
 services:
-  - docker
-
+- docker
 script:
-  - CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o certificate-controller .
-  - docker build -t certificate-controller:1.0 .
+- CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o certificate-controller
+  .
+- docker build -t easymile/certificate-controller:1.0-$(echo $TRAVIS_COMMIT | cut -c1-7) .
+- docker tag easymile/certificate-controller:1.0-$(echo $TRAVIS_COMMIT | cut -c1-7) easymile/certificate-controller:1.0-latest
+- docker login -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD
+- docker push easymile/certificate-controller:1.0-$(echo $TRAVIS_COMMIT | cut -c1-7)
```

Comment:
This is a subjective opinion perhaps but you should read the version from a dedicated file in the source tree. It isn't obvious that the version is declared in the travis.yml. You can also remove it entirely and do everything with commit hashes and latest but it will be harder to manage compatibility (if you ever want to do it).

Reply:
What would be nice is to have version tied to TRAVIS_TAG if it exists and to TRAVIS_COMMIT in the other case.